specutils/analysis/flux.py

```diff
@@ -116,3 +124,65 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None):
     ew =  dx - (line_flux / continuum)
 
     return ew.to(calc_spectrum.spectral_axis.unit)
+
+def is_continuum_near_zero(spectrum, eps=0.01):
+    """
+    Determine if the typical baseline of this spectrum is near zero. I.e., an
+    estimate of whether or not the continuum has been subtracted.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object over which the width will be calculated.
+
+    eps: float
+        The tolerance on the quantification to confirm the continuum is
+        near zero.
+
+    Returns
+    -------
+    is_near_zero: bool
+        A True if the continuum of the spectrum is with eps, False otherwise.
+
+    """
+
+    flux = spectrum.flux
+    uncertainty = spectrum.uncertainty if hasattr(spectrum, 'uncertainty') else None
+
+    # Apply the mask if it exists.
+    if hasattr(spectrum, 'mask') and spectrum.mask is not None:
+        flux = flux[~spectrum.mask]
+        uncertainty = uncertainty[~spectrum.mask] if uncertainty else uncertainty
+
+    # If the eps has units then the assumption is that we want
+    # to compare the median of the flux regardless of the
+    # existence of the uncertainty.
+    if hasattr(eps, 'unit'):
+        return np.median(flux) < eps
+
+    # If eps does not have a unit, ie it is not a quantity, then
+    # we are going to calculate based on the S/N if the uncertainty
+    # exists.
+    if uncertainty:
+        return np.median(flux / uncertainty.quantity) < eps
+    else:
+        warnings.warn('Spectrum does not appear to be continuum subtracted.', AstropyUserWarning)
+        #return np.median(flux) / median_absolute_deviation(flux) < eps
+
+def warn_spectrum_continuum_subtracted(eps=0.01, check=conf.always_check_continuum):
+    """
+    Dectorator for methods that should warn if the baseline
+    of the spectrum does not appear to be near zero.
+    """
+    def actual_decorator(function):
+        @wraps(function)
+        def wrapper(*args, **kwargs):
+            if check:
+                spectrum = args[0]
+                if not is_continuum_near_zero(spectrum, eps):
+                    warnings.warn('Spectrum does not appear to be continuum subtracted.',
```

Comment:
This should also have some text that refers back to `always_check_continuum` and tells the user exactly what they need to do to turn it off (or links to a page in the docs that explains it).

Reply:
I added a short message with a link back to the astropy configuration section.